src/workerd/io/trace-stream.c++

```diff
@@ -0,0 +1,821 @@
+#include <workerd/api/global-scope.h>
+#include <workerd/io/io-context.h>
+#include <workerd/io/trace-stream.h>
+#include <workerd/io/worker-interface.h>
+#include <workerd/jsg/jsg.h>
+#include <workerd/util/completion-membrane.h>
+#include <workerd/util/uuid.h>
+
+#include <capnp/membrane.h>
+
+namespace workerd::tracing {
+namespace {
+jsg::JsValue ToJs(jsg::Lock& js, const tracing::Attribute& attribute) {
+  auto obj = js.obj();
+  obj.set(js, "type"_kj, js.str("attribute"_kj));
+  obj.set(js, "name"_kj, js.str(attribute.name));
+
+  auto ToJs = [](jsg::Lock& js, const tracing::Attribute::Value& value) -> jsg::JsValue {
+    KJ_SWITCH_ONEOF(value) {
+      KJ_CASE_ONEOF(str, kj::String) {
+        return js.str(str);
+      }
+      KJ_CASE_ONEOF(b, bool) {
+        return js.boolean(b);
+      }
+      KJ_CASE_ONEOF(d, double) {
+        return js.num(d);
+      }
+      KJ_CASE_ONEOF(i, int32_t) {
+        return js.num(i);
+      }
+    }
+    KJ_UNREACHABLE;
+  };
```

Comment:
Since we are inside an anonymous namespace, we can just move this `ToJs` to a function in the anonymous namespace itself?